language/move-prover/bytecode-to-boogie/src/dataflow_analysis.rs

```diff
@@ -42,7 +42,8 @@ pub trait DataflowAnalysis: TransferFunctions {
     ) -> StateMap<Self::State> {
         let mut state_map: StateMap<Self::State> = StateMap::new();
         let entry_block_id = cfg.entry_block_id();
-        let mut work_list = vec![entry_block_id];
+        let mut work_list = VecDeque::new();
+        work_list.push_back(entry_block_id);
```

Comment:
I need to use a queue to make sure the blocks are visited in the right order: a block is only visited after all its predecessors have been visited.